scripts/generator.py

```diff
@@ -44,16 +36,32 @@ def main():
     ecs_helpers.make_dirs(out_dir)
     ecs_helpers.make_dirs(docs_dir)
 
-    nested, flat = intermediate_files.generate(fields, out_dir, default_dirs)
+    # To debug issues in the gradual building up of the nested structure, insert
+    # statements like this after any step of interest.
+    # ecs_helpers.yaml_dump('ecs.yml', fields)
+
+    fields = loader.load_schemas(ref=args.ref, included_files=args.include)
+    cleaner.clean(fields)
+    finalizer.finalize(fields)
+    subsets = subset_filter.load_subset_definitions(args.subset)
+    for subset in subsets:
+        subfields = subset_filter.extract_matching_fields(fields, subset['fields'])
+        intermediate_files.generate(subfields, os.path.join(out_dir, 'ecs', 'subset', subset['name']), default_dirs)
+
+    merged_subset = subset_filter.combine_all_subsets(subsets)
+    if merged_subset:
+        fields = subset_filter.extract_matching_fields(fields, merged_subset)
+
+    nested, flat = intermediate_files.generate(fields, os.path.join(out_dir, 'ecs'), default_dirs)
     if args.intermediate_only:
         exit()
 
-    csv_generator.generate(flat, ecs_version, out_dir)
     es_template.generate(flat, ecs_version, out_dir, args.template_settings, args.mapping_settings)
     beats.generate(nested, ecs_version, out_dir)
     if args.include or args.subset:
         exit()
 
+    csv_generator.generate(flat, ecs_version, out_dir)
```

Comment:
I moved this because certain subsets can break the csv generation and cause the tooling to crash (if the subset sets `enabled: false` on an object that isn't explicitly listed, like `process.thread`, then that field is no longer only an `intermediate` field but it doesn't have all the required fields for csv generation). For now I can add the necessary attributes for csv generation to a field when a subset changes the field from `intermediate: true` to `intermediate: false`, and then move the csv generation back to where it was.

Reply:
I like the concept of an `--artifacts` and think it'd be a good addition to incorporate eventually. Opened #885 to capture